src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2811,6 +2811,46 @@ exprt c_typecheck_baset::do_special_functions(
 
     return expr;
   }
+  else if(
+    identifier == CPROVER_PREFIX "overflow_minus" ||
+    identifier == CPROVER_PREFIX "overflow_mult" ||
+    identifier == CPROVER_PREFIX "overflow_plus" ||
+    identifier == CPROVER_PREFIX "overflow_shl" ||
+    identifier == CPROVER_PREFIX "overflow_unary_minus")
+  {
+    exprt overflow{identifier, typet{}, exprt::operandst{expr.arguments()}};
```

Comment:
Why not pick some appropriate type? This seems likely to trip up `typecheck_expr_binary_arithmetic` eventually

Reply:
It's the job of the type checker to work out the type of expressions (based on the type of the operands) so I think this should be fine?